[DOCS] Adds user-facing docs for the new KP logging configuration (#94993)
elastic/kibana: docs/developer/architecture/core/configuration-service.asciidoc

[[configuration-service]]
== Configuration service

> {kib} provides `ConfigService` if a plugin developer may want to support

Reviewer: ```suggestion
{kib} provides `ConfigService` for plugin developers that want to support
```
